src/test/java/com/karankumar/bookproject/backend/service/ReadingGoalServiceTest.java

```diff
@@ -19,28 +19,58 @@
 
 import com.karankumar.bookproject.annotations.IntegrationTest;
 import com.karankumar.bookproject.backend.entity.ReadingGoal;
-import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Assumptions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
 
+import static com.karankumar.bookproject.utils.ReadingGoalTestUtils.resetGoalService;
 import static org.assertj.core.api.Assertions.assertThat;
 
 @IntegrationTest
 class ReadingGoalServiceTest {
+    private final ReadingGoalService goalService;
+    private ReadingGoal existingReadingGoal;
+
+    private ReadingGoalServiceTest(@Autowired ReadingGoalService goalService) {
+        this.goalService = goalService;
+    }
+
+    @BeforeEach
+    void beforeEach(){
+        resetGoalService(goalService);
+        initReadingGoalServiceTest();
+    }
+
+    private void initReadingGoalServiceTest() {
+        this.existingReadingGoal = new ReadingGoal(20, ReadingGoal.GoalType.BOOKS);
+        goalService.save(this.existingReadingGoal);
```

Comment:
Could you please remove this `this` modifier from both statements as it is unambiguous here?

Reply:
As this was the only change needed, I've added this in